app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -409,33 +417,36 @@ public boolean onPreferenceChange(Preference preference, Object o) {
                     break;
                 case KEY_STTRACE:
                     if (((String) o).matches("[0-9a-z, ]*"))
-                        requireRestart();
+                        mRequireRestart = true;
                     else {
                         Toast.makeText(getActivity(), R.string.toast_invalid_sttrace, Toast.LENGTH_SHORT)
                                 .show();
                         return false;
                     }
                     break;
-                case "environment_variables":
+                case Constants.PREF_ENVIRONMENT_VARIABLES:
                     if (((String) o).matches("^(\\w+=[\\w:/\\.]+)?( \\w+=[\\w:/\\.]+)*$")) {
-                        requireRestart();
+                        mRequireRestart = true;
                     }
                     else {
                         Toast.makeText(getActivity(), R.string.toast_invalid_environment_variables, Toast.LENGTH_SHORT)
                                 .show();
                         return false;
                     }
                     break;
+                case Constants.PREF_USE_WAKE_LOCK:
```

Comment:
This is defined in Constants.java . We got more constants instead of duplicate strings in by my last PR's so this is a side improvement. All constants that were originally there for service and app prefs and the one of this PR are in there. The class is intended as final (= static).

Reply:
Yeah, ok, I see the code is wired in there, yet it doesn't actually need to restart syncthing.